rpc/src/client/transport/http.rs

```diff
@@ -367,8 +367,21 @@ mod sealed {
         {
             let request = self.build_request(request)?;
             let response = self.inner.request(request).await.map_err(Error::hyper)?;
+
+            tracing::debug!("Incoming response:");
+            let response_status = response.status();
+            tracing::debug!("  Status: {}", response.status());
             let response_body = response_to_string(response).await?;
-            tracing::debug!("Incoming response: {}", response_body);
+            tracing::debug!("  Body: {}", response_body);
+
+            // Successful JSON-RPC requests are expected to return a 200 OK HTTP status.
+            // Otherwise, this means that the HTTP request failed as a whole,
+            // as opposed to the JSON-RPC request returning an error,
+            // and we cannot expect the response body to be a valid JSON-RPC response.
+            if response_status != http::StatusCode::OK {
```

Comment:
There is a method `StatusCode::is_success`, but maybe success statuses that are not exactly 200 should also be seen as non-conformant.

Reply:
Yes it's my understanding from reading both the Comet source and [this spec](https://www.simple-is-better.org/json-rpc/transport_http.html), that only 200 OK responses can be interpreted as a valid JSON-RPC payload. That said, I don't know how official that spec is.

Also it does not deal with redirects, etc. but that was already the case before. Do you know if hyper takes care of those for us?